be/src/olap/rowset/segment_v2/inverted_index_writer.cpp

```diff
@@ -171,6 +171,12 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
             auto chinese_analyzer = _CLNEW lucene::analysis::LanguageBasedAnalyzer();
             chinese_analyzer->setLanguage(L"chinese");
             chinese_analyzer->initDict(config::inverted_index_dict_path);
+            auto mode = get_parser_mode_string_from_properties(_index_meta->properties());
+            if (mode == INVERTED_INDEX_PARSER_COARSE_GRANULARITY) {
+                chinese_analyzer->setMode(lucene::analysis::AnalyzerMode::Default);
```

Comment:
Normally check special case in if branches and left default to else branch.

Reply:
we want to use All mode as default